resources/javascripts/script.js

```diff
@@ -0,0 +1,145 @@
+        slides[i].classList.add("mySlidesHide");
+        slides[i].classList.remove("mySlidesShow");
+    }
+    for (i = 0; i < dots.length; i++) {
+        dots[i].className = dots[i].className.replace(" active", "");
+    }
+    slides[slideIndex - 1].classList.add("mySlidesShow");
+    slides[slideIndex - 1].classList.remove("mySlidesHide");
+    dots[slideIndex - 1].className += " active";
+};
+
+var plusSlides = (n) => {
+    showSlides(slideIndex += n);
+};
+
+var currentSlide = (n) => {
+    showSlides(slideIndex = n);
+};
+
+if (document.location.pathname == "/") {
+    showSlides(slideIndex);
+    document.getElementById("prev").addEventListener("click", function() {
+        plusSlides(-1);
+    });
+
+    document.getElementById("next").addEventListener("click", function() {
+        plusSlides(1);
+    });
+
+    document.getElementById("dot").addEventListener("click", function() {
+        currentSlide(this.getAttribute('data-attr'));
+    });
+
+    setInterval(function() { plusSlides(1); }, 1000);
+};
+
+/*-------------------------- Cart functonality --------------------------*/
+var allitemInCart = () => {
+    var url = window.location.origin + "/cart/allitem";
+    let xmlHttpReq = new XMLHttpRequest();
+    xmlHttpReq.open("GET", url, true);
+    xmlHttpReq.onload = () => {
+        if (xmlHttpReq.status == 200) {
+            let data = JSON.parse(xmlHttpReq.responseText);
+            let totalCheckoutAmt = document.getElementById("total-checkout-amt");
+            document.getElementById("item-count").innerHTML = data.itemCounter;
+            let totalCheckoutPrice = 0;
+            data.productInCart.forEach(element => {
+                totalCheckoutPrice = totalCheckoutPrice + element.totalPrice;
+            });
+            totalCheckoutAmt.innerHTML = totalCheckoutPrice;
+        }
+    };
+    xmlHttpReq.send();
+}
+
+var buyItem = (id, operation) => {
+    var url = window.location.origin + '/cart/' + id + '/' + operation;
+    let xmlHttpReq = new XMLHttpRequest();
+    xmlHttpReq.open("GET", url, true);
+    xmlHttpReq.onload = () => {
+        if (xmlHttpReq.status == 200) {
+            let data = JSON.parse(xmlHttpReq.responseText);
+            updateCart(data.itemCounter);
+        }
+    };
+    xmlHttpReq.send();
+}
+
+var updateCart = (itemCounter) => {
+    document.getElementById('item-count').innerHTML = itemCounter;
+}
+
+var updateCheckoutAmount = () => {
+    var totalAmt = document.getElementsByClassName("total-amt");
+    let totalCheckoutAmt = document.getElementById("total-checkout-amt");
+    let totalCheckoutPrice = 0;
+    for (let i = 0; i < totalAmt.length; i++) {
+        totalCheckoutPrice = totalCheckoutPrice + parseInt(totalAmt[i].innerHTML);
+    }
+    totalCheckoutAmt.innerHTML = totalCheckoutPrice;
+}
+
+if (window.location.pathname === "/product") {
+    document.getElementById("buy-item").addEventListener("click", function() {
+        let id = this.getAttribute('data-id'),

```

Comment:
use dataset instead for fetching values using getAttribute for data-attributes

Reply:
done